walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,75 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.List;
+import java.util.ListIterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> availableSlots = new ArrayList<>();
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return availableSlots;
+    }
+
+    availableSlots.add(TimeRange.WHOLE_DAY);
+    for (Event event : events) {
+      TimeRange eventTimeRange = event.getWhen();
+      boolean containsAttendee =
+              request.getAttendees().stream().anyMatch((attendee) -> event.getAttendees().contains(attendee));
```

Comment:
nit: worth extracting to a separate function. And extracting `request.getAttendees()` to a local variable.